### DISCUSSION TITLE
Suggestions for improving memory usage?

### DISCUSSION THREAD

**Discussion by chaorace** [qutebrowser/qutebrowser#5611] [Q&A] [answered]
I'm having issues with Qutebrowser consuming large amounts of memory if I have a large number of windows open.

Obviously, more windows will consume more memory, no big surprise here, but Qutebrowser seems less optimized for this than Firefox or Chromium. I routinely reach 10+ GB consumed on my Linux box!

I don't really think this is a bug per-say, I'm just looking for advice that I can follow to improve the situation. Perhaps there are some config settings that are known to cause memory issues, or maybe there's some QtWebEngine configuration flags I can try?

**Comment by The-Compiler** [comment, marked as answer]
There's some discussion about this in #1476.

**Reply by chaorace** [reply]
Thanks. I reviewed and found a couple of plausible solutions:
* Setting qt.process_model process-per-site
* Configuring qutebrowser to use jemalloc for memory allocation

There wasn't much to go on for getting jemalloc set up, but I was able to get it working on my Arch box by exporting this value for LD_PRELOAD from my bashrc:

<pre>export LD_PRELOAD=/usr/lib/libjemalloc.so</pre>.

If anyone else is trying this, obviously make sure jemalloc is installed first. Arch has it under the jemalloc package.

**Reply by chaorace** [reply]
Just wanted to followup on my results so far:
* qt.process_model process-per-site seems to work without any issues. This only helps a bit, though
* Exporting to LD_PRELOAD like I did has a lot of nasty side-effects, unfortunately. I ended up reversing that change.

As for getting qutebrowser to use jemalloc, I guess I'll end up having to set up a script on my path instead. I figure I'll get around to doing that after I finish migrating to NixOs, since it's a pretty good application for overlays. I'll make sure to share it once it's ready!
